lib/Serialization/DeserializeSIL.cpp

```diff
@@ -2566,6 +2566,12 @@ bool SILDeserializer::readSILInstruction(SILFunction *Fn,
     // Format: a type, an operand and a SILDeclRef. Use SILOneTypeValuesLayout:
     // type, Attr, SILDeclRef (DeclID, Kind, uncurryLevel), and an operand.
     unsigned NextValueIndex = 0;
+    bool isObjCDirectMethod = false;
+    if (OpCode == SILInstructionKind::ObjCMethodInst) {
+      unsigned Flags = ListOfValues[NextValueIndex];
```

Comment:
You need to increment the `SWIFTMODULE_VERSION_MINOR` in `lib/Serialization/ModuleFormat.h` if you change anything about serialization.

Also, please add a test, e.g. in `test/Serialization/Inputs/def_basic.sil`